src/tox/config/__init__.py

```diff
@@ -1591,7 +1592,7 @@ def addsubstitutions(self, _posargs=None, **kw):
             self.posargs = _posargs
 
     def getpath(self, name, defaultpath, replace=True):
-        path = self.getstring(name, defaultpath, replace=replace)
+        path = self.getstring(name, defaultpath, replace=replace, unquote_path=True)
         if path is not None:
             toxinidir = self._subs["toxinidir"]
             return toxinidir.join(path, abs=True)
```

Comment:
I was doing the unquoting here, before the join.  It worked well, but didnt feel tight enough.

There is still one aspect which may make this necessary, and at least needs a test case.
In addition to problematic chars in preset paths like `{toxinidir}`, the space or hash might appear in derived paths, such as `{toxinidir}/foo bar.txt`

Reply:
Added tests for `'{toxinidir}'/'foo bar.txt'` , and I am pretty sure that the unquoting cant occur here.